whosnext.py

```diff
@@ -167,10 +170,12 @@
 
 #--[Selection
 # Select primary presenter(s) for next meeting!
-choice = []
+choice = set()
 for i in range(presenters_per_meeting):
-    choice.append(random.choices(current_members, weights=[weights[k] for k in current_members]))
-    weights[choice[i][0]] = 0.
+    remaining_members = [member for member in current_members if member not in choice]
+    remaining_weights = [weights[member] for member in remaining_members]
+    chosen = random.choices(remaining_members, weights=remaining_weights, k=1)[0]
+    choice.add(chosen)
```

Comment:
What exactly are you fixing here? What was wrong with the prior code?

Reply:
The previous code was not a random weighted selection, just a weighted selection. It would always select the same people.